### PR TITLE
feat: add statix for nix linting

### PR DESCRIPTION
**Pull request by brianmcgee**

### CODE REVIEW

**Review by aldoborrero — CHANGES_REQUESTED** [review]
See pending comments.

Also, I think we should include a shortcut within `nix/scripts.nix` to run easily the tool, in the same way, we can do with `, fmt`.

The rest of the PR looks good enough to me.

File: nix/checks.nix [hunk 1]
```diff
@@ -5,17 +5,16 @@
...
-      #     cd source
-      #     HOME=$TMPDIR nix-linter --recursive
-      #     touch $out
-      #   '';
+      nix-lint =
+        pkgs.runCommand "statix" {
+          nativeBuildInputs = with pkgs; [statix];
+        } ''
+          cp --no-preserve=mode -r ${self} source
+          rm -rf source/.direnv # delete .direnv cache so nix-linter don't recurse it
```
[the remark this answers is not in the document]
Comment (brianmcgee) [reply nix/checks.nix]: I've added a `statix.toml` for controlling ignore patterns and simplified the checks integration

**Comment by brianmcgee** [comment]
> Also, I think we should include a shortcut within `nix/scripts.nix` to run easily the tool, in the same way, we can do with `, fmt`.

A shortcut is not needed. Just run `statix` or `nix flake check`.

**Review by aldoborrero — DISMISSED** [review]
LGTM

**Review by aldoborrero — APPROVED** [review]

**Pull request merged**